lib/util/source-code.js

```diff
@@ -252,8 +259,96 @@ SourceCode.prototype = {
         });
 
         return result;
-    }
+    },
+
+    /**
+     * Registers a fix that insert text at the specified index in the source text.
+     * The fix is not applied until applyFixes() is called.
+     * @param {int} index The 0-based index at which to insert the new text.
+     * @param {string} text The text to insert.
+     * @returns {void}
+     */
+    insertText: function(index, text) {
+        this._fixes.push({
+            range: [index, index],
+            text: text
+        });
+    },
 
+    /**
+     * Registers a fix that inserts text after the specified range in the source text.
+     * The fix is not applied until applyFixes() is called.
+     * @param {int[]} range The range to replace, first item is start of range, second
+     *      is end of range.
+     * @param {string} text The text to insert.
+     * @returns {void}
+     */
+    insertTextAfter: function(range, text) {
+        this.insertText(range[1], text);
+    },
+
+    /**
+     * Registers a fix that inserts text before the specified range in the source text.
+     * The fix is not applied until applyFixes() is called.
+     * @param {int[]} range The range to replace, first item is start of range, second
+     *      is end of range.
+     * @param {string} text The text to insert.
+     * @returns {void}
+     */
+    insertTextBefore: function(range, text) {
+        this.insertText(range[0], text);
+    },
+
+    /**
+     * Registers a fix that replaces text at the specified range in the source text.
+     * The fix is not applied until applyFixes() is called.
+     * @param {int[]} range The range to replace, first item is start of range, second
+     *      is end of range.
+     * @param {string} text The text to insert.
+     * @returns {void}
+     */
+    replaceText: function(range, text) {
+        this._fixes.push({
+            range: range,
+            text: text
+        });
+    },
+
+    /**
+     * Registers a fix that removes the specified range of text from the source.
+     * The fix is not applied until applyFixes() is called.
+     * @param {int[]} range The range to remove, first item is start of range, second
+     *      is end of range.
+     * @returns {void}
+     */
+    removeText: function(range) {
+        this._fixes.push({
+            range: range,
+            text: ""
+        });
+    },
+
+    /**
+     * Applies the cached fixes to the source code and returns the result. Does not
+     * alter the text property.
+     * @returns {string} The fixed text.
+     */
+    applyFixes: function() {
```

Comment:
Still trying to wrap my head around, so 
* how is this different then the function inside `source-code-fixer`?
* Where is this getting used?